### PR TITLE
Add case of Starting vm with file source backed NVRAM

### PR DESCRIPTION
**Pull request by chloerh**
Signed-off-by: Haijiao Zhao <haizhao@redhat.com>

Depends on:
- https://github.com/avocado-framework/avocado-vt/pull/3557

Test result:
```
PASS 1-type_specific.io-github-autotest-libvirt.vm_boot.nvram_source.file
```

### CODE REVIEW

File: libvirt/tests/src/bios/vm_boot_nvram_source.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        """
+        Test file backed source
+        """
+        os_xml = vmxml.os
+        LOG.debug('Os xml before test: %s', os_xml)
+
+        fd_file = os_xml.nvram
+        LOG.debug('Nvram file: %s', fd_file)
+
+        if not fd_file.endswith('.fd'):
```
Comment (dzhengfy) [inline libvirt/tests/src/bios/vm_boot_nvram_source.py:26]: Is this an expected checkpoint? I did not see it in the feature case.

File: libvirt/tests/src/bios/vm_boot_nvram_source.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            test.fail('Nvram file should be xxx.fd')
```
Comment (dzhengfy) [inline libvirt/tests/src/bios/vm_boot_nvram_source.py:27]: Anyway  the message should be better if we use 'Nvram file should have 'fd' as the suffix '?

Comment (dzhengfy) [reply libvirt/tests/src/bios/vm_boot_nvram_source.py:27]: If not a checkpoint, please use test.error() if it does not match.

File: libvirt/tests/src/bios/vm_boot_nvram_source.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                          {'template': os_xml.nvram_attrs['template'],
+                           'type': 'file'},
+                          'nvram_source': {'attrs': {'file': fd_file}}}
+        os_xml.del_nvram()
+        os_xml.setup_attrs(**os_nvram_attrs)
+        LOG.debug('Updated os xml: %s', os_xml)
+
+        # Reboot vm to let configuration take effect
+        virsh.reboot(vm_name)
+
```
Comment (dzhengfy) [inline libvirt/tests/src/bios/vm_boot_nvram_source.py:39]: Maybe here we can log out the vm xml for debugging.

File: libvirt/tests/src/bios/vm_boot_nvram_source.py (hunk 2 above)
Comment (chloerh) [reply libvirt/tests/src/bios/vm_boot_nvram_source.py:27]: On second thought, I'm going to remove this. It's an unnecessary check.

File: libvirt/tests/src/bios/vm_boot_nvram_source.py (hunk 3 above)
Comment (chloerh) [reply libvirt/tests/src/bios/vm_boot_nvram_source.py:37]: Done.

File: libvirt/tests/src/bios/vm_boot_nvram_source.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        fd_file = os_xml.nvram
+        LOG.debug('Nvram file: %s', fd_file)
+
+        os_nvram_attrs = {'nvram_attrs':
+                          {'template': os_xml.nvram_attrs['template'],
+                           'type': 'file'},
+                          'nvram_source': {'attrs': {'file': fd_file}}}
+        os_xml.del_nvram()
+        os_xml.setup_attrs(**os_nvram_attrs)
```
Comment (dzhengfy) [inline libvirt/tests/src/bios/vm_boot_nvram_source.py:31]: Please make sure the vm xml is updated correctly

Comment (dzhengfy) [inline libvirt/tests/src/bios/vm_boot_nvram_source.py:31]: Please make sure the vm xml is updated correctly

Comment (chloerh) [reply libvirt/tests/src/bios/vm_boot_nvram_source.py:31]: Code updated.Test running.

**Review by meinaLi — APPROVED** [review]

**Comment by chloerh** [comment]
Log:

```
...
VM xml after reboot: <domain type='kvm' id='2'>
  <name>avocado-vt-vm1</name>
  <uuid>0c370e53-e218-477b-a9a5-2bf278732ed1</uuid>
  <metadata>
    <ns0:libosinfo xmlns:ns0="http://libosinfo.org/xmlns/libvirt/domain/1.0">
      <ns0:os id="http://redhat.com/rhel/9.1"/>
    </ns0:libosinfo>
  </metadata>
  <memory unit='KiB'>2097152</memory>
  <currentMemory unit='KiB'>2097152</currentMemory>
  <vcpu placement='static'>2</vcpu>
  <resource>
    <partition>/machine</partition>
  </resource>
  <os>
    <type arch='x86_64' machine='pc-q35-rhel9.0.0'>hvm</type>
    <loader readonly='yes' type='pflash'>/usr/share/edk2/ovmf/OVMF_CODE.secboot.fd</loader>
    <nvram template='/usr/share/edk2/ovmf/OVMF_VARS.secboot.fd' type='file'>
      <source file='/var/lib/libvirt/qemu/nvram/avocado-vt-vm1_VARS.fd'/>
    </nvram>
    <boot dev='hd'/>
  </os>
  <features>
...

```

**Review by dzhengfy — APPROVED** [review]

**Pull request merged**